schedulearn-frontend/src/server-components/TopicList.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import dotnetify from "dotnetify";
+import { Item } from "src/components/NestedList/NestedListItem";
+import { NestedList } from "src/components/NestedList/NestedList";
+dotnetify.hubServerUrl = "http://localhost:5000";
+
+interface TopicListState {
+  RootTopic?: FullTopic;
+}
+interface TopicListProps {
+  onItemClick?(item: FullTopic, itemIndex: string): void;
+  width?: number;
+  selectedItemIndex?: string;
+}
+
+export interface FullTopic extends Item<FullTopic>{
+  Id: number;
+  Description: string;
+  ParentTopicId: number;
+}
+
+export default class TopicList extends React.Component<TopicListProps, TopicListState> {
+  constructor(props: TopicListProps) {
+    super(props);
+    dotnetify.react.connect("TopicList", this);
+    this.state = { };
+  }
+
+  render(): React.ReactNode {
+    if(this.state.RootTopic === undefined)
+    {
+      return(
+        <div>Loading stuff</div>      
```

Comment:
Change this to look better for user. Perhaps pass an item to nested list, that has empty subitems and label that signifies loading, for example:
```
const topic = this.state.RootTopic;
if (this.state.RootTopic === undefined) {
  return (<NestedList item={{Label: "Loading...", SubItems: []}}/>);
}
...
```

I will consider adding a placeholder option for NestedList to make this more clean and avoid creating a second list later on.